src/library/scala/collection/mutable/ListBuffer.scala

```diff
@@ -114,6 +114,15 @@ class ListBuffer[A]
     this
   }
 
+  // Overridden for performance (to avoid virtual call to addOne
+  override final def addAll(xs: IterableOnce[A]): this.type = {
+    val it = xs.iterator
+    while (it.hasNext) {
+      addOne(it.next())
```

Comment:
we could manually inline this
How much do we sav by avoiding repeated calls to ensureUnaliased and the field accesses?
The first iteration of the loop is special

Reply:
Good point. It seems we can shave a further 7% with that approach. I've pushed an additional commit.